examples/translate-all/src/index.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * Custom sidebar widget to translate all content.
+ * Displays a “Translate all content” button in the sidebar.
+ */
+
+import Localizer from './localizer'
+import NonLocalizedLocalizablesGenerator from './localizables-generator'
+
+const $ = document.querySelector.bind(document)
+
+window.contentfulWidget.init(initWidget)
+
+function initWidget (cfApi) {
+  cfApi.window.startAutoResizer()
+
+  const contentTypeId = cfApi.entry.getSys().contentType.sys.id
+  const translateButton = $('#trall-btn-translate')
+  let localesGenerator
+
+  cfApi.space.getContentType(contentTypeId).then(function (contentType) {
+    const fieldTypes = getFieldTypesMapFromContentType(contentType)
+    localesGenerator =
+      new NonLocalizedLocalizablesGenerator(
+        cfApi.entry, fieldTypes, cfApi.locales.default)
+
+    translateButton.disabled = false
+  })
+
+  translateButton.addEventListener('click', () => {
+    translateButton.classList.add('cf-is-loading')
+    translateButton.disabled = true
+
+    newLocalizer().localize(localesGenerator.generateLocalizables())
+      .then(releaseButton)
+      .catch(releaseButton)
```

Comment:
If the promise implementation supports it we should use `.finally(releaseButton)`